src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -1005,9 +1009,10 @@ private Expression AddOrderByQueryForSource(Expression source, OrderByClause ord
             //      (ElementType element) => new Wrapper { }
             LambdaExpression selector = Expression.Lambda(projection, element);
 
+            Expression modifiedSource = source;
             if (orderByClause != null)
             {
-                source = AddOrderByQueryForSource(source, orderByClause, elementType);
```

Comment:
Weird, save the source to a new variable fixes the issue?

Reply:
There's still one reference to "source" in the null propagation code. That's the fix - making sure the null check uses the original source instead of the modified.